flocker/control/test/test_protocol.py

```diff
@@ -493,6 +506,33 @@ def setUp(self):
         self.protocol = ControlAMP(self.reactor, self.control_amp_service)
         self.client = LoopbackAMPClient(self.protocol.locator)
 
+    def test_connection_stays_open_on_activity(self):
+        """
+        The AMP connection remains open when communication is received at
+        any time up to the timeout limit.
+        """
+        self.protocol.makeConnection(StringTransportWithAbort())
+        advance_some(self.reactor)
+        self.client.callRemote(NoOp)
+        self.assertFalse(self.protocol.transport.aborted)
+        self.reactor.advance(PING_INTERVAL.seconds * 1.9)
+        # This NoOp will reset the timeout.
+        self.client.callRemote(NoOp)
+        self.reactor.advance(PING_INTERVAL.seconds * 1.9)
+        self.assertEqual(self.protocol.transport.aborted, False)
```

Comment:
:question: 

You could do single assert at end, which is better since it centralizes the decision of success vs. failure:

```
initially_aborted = self.protocol.transport.aborted
...
later_aborted = self.protoco..transport.aborted
self.assertEqual(dict(initially=initially_aborted, later=later_aborted),
                           dict(initally=False, later=False))
```